addon/types/index.ts

```diff
@@ -3,10 +3,15 @@ export type WaiterName = string;
 export interface IWaiter {
   name: WaiterName;
   waitUntil(): boolean;
-  debugInfo(): unknown[] | undefined | void;
+  debugInfo(): unknown;
```

Comment:
Hmm, don't we want to ensure it always returns an array (which would mean `unknown[]`)?

When this is used by @ember/test-helper's I'd rather be able to make the safe assumption that it always returns an array...

Reply:
If people define their own `IWaiter`, they may want to return their own specific type or format from `debugInfo`. If we require them to return an array, that's constraining them to a specific shape of data. I'm not sure we want to do that.

Reply:
I'm unconvinced. Lets keep an eye on the work that this places on the `@ember/test-helpers` integration and see if we want to revisit...